metricbeat/module/kubernetes/container/data.go

```diff
@@ -39,8 +39,9 @@ func eventMapping(content []byte, perfMetrics *util.PerfMetricsCache, logger *lo
 	}
 
 	node := summary.Node
-	nodeCores := perfMetrics.NodeCoresAllocatable.Get(node.NodeName)
-	nodeMem := perfMetrics.NodeMemAllocatable.Get(node.NodeName)
+	nodeMetricOwner := util.GetMetricOwner(node.NodeName, util.NODE_METRIC_PREFIX)
+	nodeCores, _ := metricsStorage.GetMetricWithDefault(nodeMetricOwner, util.NODE_CORES_ALLOCATABLE_METRIC, 0.0)
+	nodeMem, _ := metricsStorage.GetMetricWithDefault(nodeMetricOwner, util.NODE_MEMORY_ALLOCATABLE_METRIC, 0.0)
```

Comment:
We return the value of `exists` but we never actually check it. What is the purpose of this value then? Also what will happen if value does not actually exist? Do we just continue? In any case we should be declarative about whatever we do in that case.

Reply:
logically I would like the API to tell you if the value it returned is from the storage or from the default. Having said that, I wasn't checking the result but I was planning to add some logs to report this